lab_3_generate_by_ngrams/main.py

```diff
@@ -410,9 +470,22 @@ def continue_sequence(
 
         In case of corrupt input arguments or unexpected behaviour of methods used return None.
         """
+        if not isinstance(sequence, tuple) or not sequence \
+                or not isinstance(next_tokens, list) or not next_tokens \
+                or not isinstance(sequence_candidates, dict) \
+                or not sequence_candidates \
+                or not len(next_tokens) <= self._beam_width \
+                or sequence not in sequence_candidates:
+            return None
+        for token in next_tokens:
+            new_sequence = sequence + (token[0],)
+            new_value = sequence_candidates[sequence] - math.log(token[1])
+            sequence_candidates[new_sequence] = new_value
+        del sequence_candidates[sequence]
+        return sequence_candidates
```

Comment:
Beware of mutation! The method should not change the object outside itself, yours does

Reply:
I think I have fixed it, but maybe I am mistaken. If so, cound you please provide me with a hint how to solve it